tools/generate_hotkeys.py

```diff
@@ -13,34 +17,135 @@
 # Name of the script
 SCRIPT_NAME = PurePath(__file__).name
 
-categories = {
-    category: {
-        item["help_text"]: item["keys"]
-        for item in KEY_BINDINGS.values()
-        if item["key_category"] == category
+# Exclude keys from duplicate keys checking
+KEYS_TO_EXCLUDE = ["q", "e", "m", "r"]
+
+
+def main(check_only: bool = False) -> None:
+    if check_only:
+        lint_hotkeys_file()
+    else:
+        generate_hotkeys_file()
+
+
+def read_help_categories() -> Dict[str, Dict[str, List[str]]]:
+    """
+    Get all help categories from keys.py
+    """
+    categories = {
+        category: {
+            item["help_text"]: item["keys"]
+            for item in KEY_BINDINGS.values()
+            if item["key_category"] == category
+        }
+        for category in HELP_CATEGORIES.keys()
     }
-    for category in HELP_CATEGORIES.keys()
-}
+    return categories
+
 
-with open(TEMP_OUTPUT_FILE, "w") as mdFile:
-    mdFile.write(
-        f"<!--- Generated automatically by tools/{SCRIPT_NAME} -->\n"
-        "<!--- Do not modify -->\n\n# Hot Keys\n"
+def various_key_combination(key_combinations_list: List[str]) -> str:
+    """
+    Returns list of key combinations.
+    """
+    various_key_combinations = " / ".join(
+        [
+            " + ".join([f"<kbd>{key}</kbd>" for key in key_combination.split()])
+            for key_combination in key_combinations_list
+        ]
     )
-    for action in categories.keys():
+    return various_key_combinations
+
+
+def compare_temp_hotkeys_file(generate_flag: bool = False) -> None:
+    """
+    To compare temp_hotkeys and hotkeys file.
+    """
+    if filecmp.cmp(ORIGINAL_OUTPUT_FILE, TEMP_OUTPUT_FILE, shallow=False):
+        print("Hot Keys already in sync with config/keys.py")
+    else:
+        print("Hot Keys not in sync with config/keys.py")
+        if not generate_flag:
+            exit(1)
```

Comment:
Here why do we want to generate the `ORIGINAL_OUTPUT_FILE`, i.e., not exit, even though it's not in sync?